lightning/src/ln/channelmanager.rs

```diff
@@ -3578,7 +3579,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				hash_map::Entry::Vacant(_) => return Err(MsgHandleErrInternal::send_err_msg_no_close("Failed to find corresponding channel".to_owned(), msg.channel_id))
 			}
 		};
-		self.claim_funds_internal(channel_lock, htlc_source, msg.payment_preimage.clone(), Some(forwarded_htlc_value), false);
+		let payment_hash = PaymentHash(Sha256::hash(&msg.payment_preimage.0).into_inner());
```

Comment:
Hmm, yuck, so in the common case of forwarding an HTLC we has the preimage an extra time here (in addition to doing it in channel) and then don't even use it in claim_funds_internal. We could either pass the payment hash to the channel or move it to `claim_funds_internal` in the `OutboundRoute` block.

Reply:
I've done the latter. Move the `payment_hash` creation to `claim_funds_internal` in  the `OutboundRoute` block.